src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -1648,15 +1648,15 @@ SubqueryPushdownMultiNodeTree(Query *originalQuery)
 	 */
 	if (extendedOpNode->groupClauseList != NIL)
 	{
-		extendedOpNode->targetList =
-			(List *) expression_tree_mutator((Node *) extendedOpNode->targetList,
-											 AddAnyValueAggregates,
-											 extendedOpNode->groupClauseList);
+		extendedOpNode->targetList = (List *) WrapUngroupedVarsInAnyValueAggregate(
+			(Node *) extendedOpNode->targetList,
+			extendedOpNode->groupClauseList,
+			extendedOpNode->targetList);
 
 		extendedOpNode->havingQual =
-			expression_tree_mutator((Node *) extendedOpNode->havingQual,
-									AddAnyValueAggregates,
-									extendedOpNode->groupClauseList);
+			WrapUngroupedVarsInAnyValueAggregate((Node *) extendedOpNode->havingQual,
```

Comment:
Hmz, seems there is an issue:

```sql
GROUP BY a.key, a.value ||'_append'
HAVING length(a.value) < length(a.value || '_append')
ORDER BY 1;
ERROR:  column "worker_subquery.worker_column_2" must appear in the GROUP BY clause or be used in an aggregate function
CONTEXT:  while executing command on localhost:57637
```

Reply:
We probably shouldn't check for equal(..) in the case of HAVING.

Reply:
Fixed that way now.

Reply:
Ok, that works I think. I tested with some more expressions in the HAVING, and looks good.

In general, we should probably test with primary keys more often, see the worker query:
```SQL
EXPLAIN ANALYZE
SELECT
  count(*)
FROM items a LEFT JOIN other_items b ON (a.key = b.key)
...
NOTICE:  issuing EXPLAIN (ANALYZE TRUE, VERBOSE FALSE, COSTS TRUE, BUFFERS FALSE, TIMING TRUE, SUMMARY TRUE, FORMAT TEXT) SELECT count(*) AS count FROM (SELECT FROM (public.items_102074 a LEFT JOIN (SELECT NULL::text AS key, NULL::text AS value WHERE false) b(key, value) ON ((a.key OPERATOR(pg_catalog.=) b.key)))) worker_subquery
...
```
Where the right table is replaced with empty subquery. We might have some other bugs that we couldn't spot because quite a lot of the regression test tables don't have any constraints.